package/MDAnalysis/analysis/rms.py

```diff
@@ -574,56 +575,46 @@ def run(self, start=None, stop=None, step=None, progout=10,
             which means that the trajectory would be read until the end.
         step : int (optional)
             step between frames, default None becomes 1.
-        progout : int (optional)
-            number of frames to iterate through between updates to progress
-            output; ``None`` for no updates [10]
+        weights : str / array_like (optional)
+            used weights. If ``'mass'`` use masses of atomgroup, it ``None`` use uniform weights.
         verbose : bool (optional)
-            if ``False``, suppress all output (implies *progout* = ``None``)
-            [``True``]
-
-        References
-        ----------
-        .. [Welford1962] B. P. Welford (1962). "Note on a Method for
-           Calculating Corrected Sums of Squares and Products." Technometrics
-           4(3):419-420.
-
-        .. deprecated:: 0.16
-           The keyword argument *quiet* is deprecated in favor of *verbose*.
+            if ``False``, suppress all output
         """
-        traj = self.atomgroup.universe.trajectory
-        start, stop, step = traj.check_slice_indices(start, stop, step)
-        sumsquares = np.zeros((self.atomgroup.n_atoms, 3))
-        means = np.array(sumsquares)
-
-        verbose = _set_verbose(verbose, quiet, default=True)
-        if not verbose:
-            progout = None
-
-        # set up progress output
-        if progout:
-            percentage = ProgressMeter(self.atomgroup.universe.trajectory.n_frames,
-                                       interval=progout)
-        else:
-            percentage = ProgressMeter(self.atomgroup.universe.trajectory.n_frames,
-                                       verbose=False)
+        super(RMSF, self).__init__(atomgroup.universe.trajectory, **kwargs)
+        self.atomgroup = atomgroup
+        if not isinstance(weights, (list, tuple, np.ndarray)) and weights == 'mass':
+            weights = self.atomgroup.masses
+        self.weights = weights
+
+    def run(self, start=None, stop=None, step=None, progout=None,
+            verbose=None, quiet=None):
+        if np.any([el is not None for el in (start, stop, step, progout, quiet)]):
+            warnings.warn("run arguments are deprecated. Please pass them at "
+                          "class construction. These options will be removed in 0.17.0",
+                          category=DeprecationWarning)
+            if quiet is not None:
```

Comment:
It would be better here to use `mda.lib.log._set_verbose`. That would do the same test as this, plus it would test the case where both *quiet* and *verbose* are provided. But most importantly, it will help latter on to find all the places to change when we will remove *quiet*.